packages/ipfs-core/src/components/storage.js

```diff
@@ -43,6 +43,8 @@ class Storage {
 
     const { peerId, keychain, isNew } = await loadRepo(repo, options)
 
+    // TODO: throw error?
```

Comment:
Seems like throw is not a great option here, if we keychain does not exists we should probably create it ? Otherwise I don't see how user is supposed to use this without getting an error.

Reply:
As far as I understand, Keychain will always exist at this point as this is only used after start. The problem is that the types are not sure about it.